### DISCUSSION TITLE
v0.15.0: Alpha release #6

### DISCUSSION THREAD

**Discussion by davmac314** [davmac314/dinit#58] [Announcements]
This release contains bug fixes and new features.

Changes:
 * New _dinit-monitor_ utility can be used to perform notifications (via an external command, such as _notify-send_) when a service changes state. This is primarily intended for users running dinit in "user mode" rather than for system instances of dinit, although it works in either case. See the _dinit-monitor(8)_ manual page for details.
 * Support on Linux for running service processes in a cgroup, via the `runs-in-cgroup` service   setting. Cgroups must be mounted/created outside of dinit (eg via an early service), there is no support for this in dinit itself. See the _dinit-service(5)_ manual page for details.
 * dinitctl's `enable`, `disable` and `rmdep` subcommands now give better feedback in various situations. For example, `dinitctl disable` will warn if a disabled service has other dependents (which may cause it to start anyway).
 * When parse errors occur in service description files, the relevant line number should now be reported (by dinitcheck, or in the dinit log).
 * Environment variable substitution in command lines is now performed for the `stop-command` setting in exactly the same way as for the `command` setting. Previously, by oversight, substitution was not performed for the `stop-command` setting.
 * Fix: start timeout was not being honoured for service restarts.

<hr /><em>This discussion was created from the release <a href='https://github.com/davmac314/dinit/releases/tag/v0.15.0'>v0.15.0: Alpha release #6</a>.</em>

**Comment by mobin-2008** [comment]
Hello.
I have some questions about this release:
1. How does dinit find out about the availability of cgroups? @davmac314 You said that dinit does not perform any of the tasks associated with creating and organizing cgroups. It only provides a feature that allows a service to run in a specific cgroup. Based on this, I think this option should be added to dinit to specify exactly which service cgroups is available with:
````
/etc/dinit.d/cgroups
------------------------------
type = scripted
command = /etc/dinit.d/scripts/cgroup
restart = false
depends-on = pseudofs

options = starts-cgroups
````
Maybe this is not necessary at all! Just write a `depends-on = cgroups` for any program that needs` runs-in-cgroup = `

2. Why not provide a checksum for each release?

regards

**Comment by davmac314** [comment]
> How does dinit find out about the availability of cgroups?

It doesn't; a service which needs to run in a cgroup should either depend on another service which creates the cgroup, or the cgroup should be created outside of dinit entirely (e.g by a pre-init process that runs from the initrd).

If the cgroup doesn't exist when the service starts, the service won't start.

> Maybe this is not necessary at all! Just write a `depends-on = cgroups` for any program that `needsruns-in-cgroup =`

That is definitely one option!

> Why not provide a checksum for each release?

I may consider signing release tarballs in future.

**Comment by mobin-2008** [comment]
> It doesn't; a service which needs to run in a cgroup should either depend on another service which creates the cgroup, or the cgroup should be created outside of dinit entirely (e.g by a pre-init process that runs from the initrd).

> If the cgroup doesn't exist when the service starts, the service won't start.

> That is definitely one option!

Ok. thanks for reply.

> I may consider signing release tarballs in future.

I do not think it is necessary to sign tarball. Only one sha256sum is enough.
my `dinit-0.15.0.tar.xz` sha256sum:
`7ba4af9148885dddf1c0dad79d9a9bde9c3e296949a7da81dcb25ef35ea8dcfb  dinit-0.15.0.tar.xz`
